src/binaryManager.ts

```diff
@@ -93,33 +100,76 @@ async function getConfiguredMirrordBinary(): Promise<string | null> {
 }
 
 /**
- * Downloads mirrord binary (if needed) and returns its path
- */
-export async function getMirrordBinary(): Promise<string> {
+ * Toggles auto-update of mirrord binary.
+ * Criteria for auto-update:
+ * - Auto-update is enabled by default
+ * - if mirrord binary path is mentioned in workspace settings, then that is used
+ * - if a version is specified, that version is downloaded
+ * - if auto-update is enabled, then latest supported version is downloaded
+ * - if auto-update is disabled, any local mirrord binary is used
+ * 
+ * @param extensionActivate If true, then a global state is set so that any changes to the workspace settings pertaining
+ * to mirrord binary auto-update will prompt the user to reload the window.
+ * @returns Path to mirrord binary
+*/
+export async function getMirrordBinary(extensionActivate?: boolean): Promise<string> {
     const configured = await getConfiguredMirrordBinary();
+
     if (configured) {
+        vscode.window.showInformationMessage(`Using mirrord binary specified in settings: ${configured}`);
         return configured;
     }
+    const latestVersion = await getLatestSupportedVersion(10000);
```

Comment:
Before, the code had logic where if there was a local binary already, the timeout was shorter. Why was it removed?

Reply:
I dont get the reasoning behind the timeout, was gonna ask Michal tmrw

Reply:
Originally that was Aviram's code, I didn't analyse. At some point I was just moving stuff around

Reply:
I have removed it since we were not using it